### DISCUSSION TITLE
[Refit] Is there any way to report the progress of an upload?

### DISCUSSION THREAD

**Discussion by SuperJMN** [reactiveui/refit#1487] [Q&A] [answered]
I have a method to upload a file to the API:

    [Multipart]
    [Post("/{path}")]
    Task Upload(string path, StreamPart streamPart);

I'd like to implement progress reporting, but I'm completely stuck. Once I call the method, I lost control over the stream that is being sent.

Is there any way to indicate some kind of callback to let my application know the progress of the upload? 
Thanks.

**Comment by anaisbetts** [comment, marked as answer]
I think the thing you'd have to do right now is to wrap the Stream in like a "ProgressNotifyingStream" that you make, to see what % of the stream has been read
